src/browser/new-browser.ts

```diff
@@ -254,7 +260,18 @@ export class NewBrowser extends Browser {
             capabilities[capabilitySettings.capabilityName]!.binary ||= executablePath;
         }
 
-        return capabilities;
+        const filteredCapabilities: WebdriverIO.Capabilities = Object.keys(capabilities)
+            .filter(capabilityName => {
+                const isW3CCapability = W3C_CAPABILITIES.includes(capabilityName);
+                const isVendorSpecificCapability = VENDOR_CAPABILITIES[browserNameW3C].includes(capabilityName);
+
+                return isW3CCapability || isVendorSpecificCapability;
+            })
+            .reduce((acc, capabilityName) => {
+                return set(acc, [capabilityName], capabilities[capabilityName as keyof WebdriverIO.Capabilities]);
+            }, {});
+
+        return filteredCapabilities;
```

Comment:
I suggest to write a message if some capabilities were omitted, because I think we might stumble on this in the future in circumstances we can't foresee now. Something like "Dropping non-standard capabilities due to running in local mode". Perhaps this should be in debug logs.